### PR TITLE
Support transformers 4.42

### PR DESCRIPTION
**Pull request by fxmarty**

### CODE REVIEW

**Comment by fxmarty** [comment]
will need to patch mistral https://github.com/huggingface/transformers/pull/31696

**Comment by fxmarty** [comment]
Transformers CI looks broken.

**Comment by fxmarty** [comment]
Failing tests are unrelated

File: optimum/exporters/onnx/model_configs.py [hunk 1]
```diff
@@ -312,6 +313,12 @@ class MistralOnnxConfig(TextDecoderWithPositionIdsOnnxConfig):
     DUMMY_PKV_GENERATOR_CLASS = MistralDummyPastKeyValuesGenerator
     NORMALIZED_CONFIG_CLASS = NormalizedTextConfig.with_args(num_key_value_heads="num_key_value_heads", allow_new=True)
 
+    def patch_model_for_export(
+        self, model: Union["PreTrainedModel", "TFPreTrainedModel"], model_kwargs: Optional[Dict[str, Any]] = None
+    ) -> "ModelPatcher":
+        print("IN patch_model_for_export")
```
Comment (IlyasMoutawwakil) [inline optimum/exporters/onnx/model_configs.py:319]: should be removed

File: optimum/exporters/onnx/model_patcher.py [hunk 2]
```diff
@@ -931,3 +950,182 @@ def patched_forward(
...
+
+            if _transformers_version >= version.parse("4.36"):
+                AttentionMaskConverter._unmask_unattended = _unmask_unattended_patched_staticmethod
+
+        if _transformers_version >= version.parse("4.36"):
+            patch_everywhere(
+                "_prepare_4d_causal_attention_mask_for_sdpa", _prepare_4d_causal_attention_mask_for_sdpa_patched
+            )
+
+        print("self._model in enter", self._model)
```
Comment (IlyasMoutawwakil) [inline optimum/exporters/onnx/model_patcher.py:1080]: same

**Review by IlyasMoutawwakil — APPROVED** [review]
some prints to remove but otherwise LGTM, thanks for the nice patch of mistral

**Comment by IlyasMoutawwakil** [comment]
@fxmarty do you think it makes sense to also remove the constraint on numpy version as it's handled by transformers now ?

File: .github/workflows/test_onnx.yml [hunk 3]
```diff
@@ -27,7 +27,7 @@ jobs:
         python-version: ${{ matrix.python-version }}
     - name: Install dependencies
       run: |
-        pip install .[tests,onnxruntime,exporters-tf]
+        pip install .[tests,onnxruntime,exporters]
```
Comment (echarlaix) [inline .github/workflows/test_onnx.yml:30]: we could even have instead
```suggestion
        pip install .[tests,exporters]
```

**Review by echarlaix — APPROVED** [review]
LGTM, thanks @fxmarty

**Pull request merged**